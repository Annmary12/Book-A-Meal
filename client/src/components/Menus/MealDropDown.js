import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MealCard from '../MealCard/MealCard.js';
import styles from './menus.scss';
import upArrow from '../../../assets/img/up-arrow.svg';
import downArrow from '../../../assets/img/down-arrow.svg';
import OrderModal from '../Orders/OrderModal/OrderModal.js';

/**
 * @class MealDropDown
 * @param {Object} event Javascript event object
 */
class MealDropDown extends Component {
  static propTypes = {
    meal: PropTypes.objectOf(PropTypes.oneOfType([
      PropTypes.string, PropTypes.number
    ])).isRequired
  }

  state = {
    droppedDown: false,
    showOrderModal: false
  }

  handleOpenOrderModal = (event) => {
    event.stopPropagation();

    this.setState({
      showOrderModal: true
    });
  }

  handleCloseOrderModal = () => {
    this.setState({
      showOrderModal: false
    });
  }

  toggleDropDown = () => {
    this.setState(prevState => ({ droppedDown: !prevState.droppedDown }));
  }

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    const { meal } = this.props;
    const arrow = this.state.droppedDown ? upArrow : downArrow;

    return (
      <div className={styles.meal}>
        <div
          aria-hidden
          role="button"
          className={styles.mealTitle}
          onClick={this.toggleDropDown}
        >
          * {meal.title}
          <div className={styles.leftSection}>
            <button className={styles.orderBtn} onClick={this.handleOpenOrderModal}>Order</button>
            <img className={styles.img} src={arrow} alt="down-arrow" />
          </div>
        </div>
        {this.state.droppedDown &&
          <MealCard
            className={styles.mealCard}
            meal={meal}
          />
        }
        <OrderModal
          isOpen={this.state.showOrderModal}
          handleClose={this.handleCloseOrderModal}
          meal={this.props.meal}
        />
      </div>
    );
  }
}

export default MealDropDown;
