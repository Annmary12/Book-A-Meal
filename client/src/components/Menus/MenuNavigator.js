import React from 'react';
import PropTypes from 'prop-types';
import prevIcon from '../../../assets/img/prev.svg';
import nextIcon from '../../../assets/img/next.svg';
import styles from './menus.scss';

const MenuNavigator = props => (
  <div className={styles.navigators}>
    {
      props.prevExists() &&
      <button onClick={props.goToPrevMenu}>
        <img src={prevIcon} alt="prev" />
        Prev
      </button>
    }
    {
      props.nextExists() &&
      <button onClick={props.goToNextMenu} className={styles.right}>
        Next
        <img src={nextIcon} alt="next" />
      </button>
    }
  </div>
);

MenuNavigator.propTypes = {
  goToNextMenu: PropTypes.func.isRequired,
  goToPrevMenu: PropTypes.func.isRequired,
  prevExists: PropTypes.func.isRequired,
  nextExists: PropTypes.func.isRequired
};

export default MenuNavigator;
