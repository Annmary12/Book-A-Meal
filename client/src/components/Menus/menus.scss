@import '../../../assets/scss/_variables';

.menu {
  min-height: 399px;
  width: 80%;
  display: flex;
  flex-direction: column;
  margin: auto;
  border-radius: 2px;
  margin-top: 2px;

  @media only screen and (max-width: 600px) {
    background: transparent;
    box-shadow: none;
    width: 100%;
    display: flex;
    flex-direction: column;
    margin: auto;
    margin-top: 54px;
  }
}

.menuTop{
  height: 52px;
  width: 101%;
  background-color: white;
  background: linear-gradient(to bottom, #963f0d 1%,#f7c485 30%,#f9b868 70%,#963f0d 100%);
  box-shadow: $card-shadow;

  @media only screen and (max-width: 600px) {
    height: 0px;
  }
}

.scroll-bottom{
  margin-top: auto;
  height: 58px;
  width: 102%;

  background-color: white;
  background: linear-gradient(to bottom, #963f0d 1%,#f7c485 30%,#f9b868 70%,#963f0d 100%);
  box-shadow: 0 1px 127px rgba(0, 0, 0, 0.93), 0 1px 2px rgba(0,0,0,0.24);
}

.menuBody{
  height: 100%;
  padding-left: 43px;
  padding-right: 43px;
  padding: 0px;
  background-color: white;
  width: 92%;
  /* text-align: center; */
  margin: auto;
  margin-top: 0px;
  box-shadow: $card-shadow;
  border-radius: 5px;
  padding: 7px;

  @media only screen and (max-width: 600px) {

  }

  @media only screen and (min-width: 1200px) {
    width: 70%;
  }
}

.menuHeader{
  font-family: Poppins, calibri;
  font-size: 26px;
  margin-top: 10px;
  font-weight: bold;
  text-align: center;
  margin-bottom: 35px;
  margin-top: 20px;

  @media only screen and (max-width: 600px) {
    font-size: 23px;
    margin-bottom: 20px;
    margin-top: 19px;
  }
}

.catererName{
  font-family: Poppins, calibri;
  font-size: 22px;
  text-align: center;
  margin-bottom: 10px;


  @media only screen and (max-width: 600px) {
    font-size: 19px;
  }
}

.categoryTitle {
  margin: 0;
   font-weight: 400;
   font-size: 48px;

   font-family: "Dancing Script",cursive, calibri;
   font-size: 27px;
}

.scroll-body li{
  display: flex;
  margin-bottom: 6px;
}
.handwritten{
	font-family: handWritten;
	font-weight: 300;
}

.mealList{
  font-family: Poppins, calibri;
  margin-top: 0px;
  font-size: 16px;
  list-style: none;
  padding-left: 0px;

  .meal{
    border: 1px solid #c5c5c5;
    padding-bottom: 9px;
    padding-top: 9px;
    margin-bottom: 1px;
    border-radius: 5px;
    background-color: #f7f7f7;

    .mealTitle{
      padding-left: 3px;
    }
    .img{
      width: 20px;
      height: 16px;
      margin-right: 7px;
    }

    .leftSection{
      margin-left:auto;
    }
  }


}

.mealTitle{
  display: flex;
  cursor: pointer;
  height: 100%;
  width: 100%;
  background: transparent;
  border: none;
  outline: none;
}

.mealCard{
  margin-bottom: 12px;
  border: 1px solid #e6e6e6;
  border-radius: 5px;
  width: 271px;
  box-shadow: none;
  margin-top: 10px;

  :global{
    animation: fadeInDown .05s ease-in-out;
  }

  &:hover{
    box-shadow: none;
  }
}

.navigators{
  display: flex;
  padding: 10px;
  width: 90%;
  margin-left: auto;
  margin-right: auto;

  .right{
    margin-left: auto;
  }

  button {
    background-color: $navigator-btn-bg-color;
    border: 2px solid darken($navigator-btn-bg-color, 10%);
    border-radius: 4px;
    outline: none;
    cursor: pointer;

    &:hover{
      background-color: darken($navigator-btn-bg-color, 10%);
      box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
    }

    @media only screen and (max-width: 600px) {
      font-size: 15px;
    }
  }

  img{
    width: 10px;
  }

  @media only screen and (min-width: 1200px) {
    width: 70%;
  }
}

.flipIn{
  :global{
    animation: flipInY .4s ease-in;
  }
}

.flipOut{
  :global{
    animation: flipOutY .4s ease-in;
  }
}


.orderBtn{
	background-color: $order-btn-bg-color;
  border: 2px solid darken($order-btn-bg-color, 10%);
  color: white;
  border-radius: 5px;
  padding-left: 9px;
  padding-right: 9px;
  font-size: 17px;
  margin-right: 13px;
  text-decoration: none;
  cursor: pointer;

  @media only screen and (max-width: 600px) {
    font-size: 14px;
    margin-right: 2px;
  }

  &:hover{
    background-color: darken($order-btn-bg-color, 10%);
  }
}

.slideInUp {
  :global{
    animation: slideInUp .3s ease-in-out;
  }
}
.noMenu{
  @extend .menuBody;
  @extend .slideInUp;

  height: 100%;
  background-color: white;
  width: 92%;
  margin: auto;
  margin-top: 0px;
  box-shadow: $card-shadow;
  border-radius: 5px;
  padding: 15px;
}
