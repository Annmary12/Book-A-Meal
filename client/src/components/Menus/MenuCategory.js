import React from 'react';
import PropTypes from 'prop-types';
import MealDropDown from './MealDropDown.js';
import styles from './menus.scss';

const MenuCategory = ({ category }) => (
  <div>
    <h3 className={styles.categoryTitle}>-{category.title}</h3>
    <ul className={styles.mealList}>
      {
        category.meals.map(meal => (
          <li key={meal.id}>
            <MealDropDown meal={meal} />
          </li>
          ))
      }
    </ul>
  </div>
);

MenuCategory.propTypes = {
  category: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array
  ])).isRequired
};

export default MenuCategory;
