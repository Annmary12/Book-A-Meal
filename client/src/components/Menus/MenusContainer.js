import { connect } from 'react-redux';
import { getMenusForTheDay } from '../../actions/menusForTheDay.js';
import Menus from './Menus.js';

const mapStateToProps = state => ({
  loading: state.menusForTheDay.isFetching,
  menus: state.menusForTheDay.menus,
  loggedIn: state.auth.loggedIn
});

export default connect(mapStateToProps, { getMenusForTheDay })(Menus);
