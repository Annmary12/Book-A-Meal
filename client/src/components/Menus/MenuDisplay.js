import React, { Component } from 'react';
import PropTypes from 'prop-types';
import isEmpty from 'lodash/isEmpty';
import NoMenu from '../NoMenu/NoMenu.js';
import MenuCategory from './MenuCategory.js';
import MenuNavigator from './MenuNavigator.js';
import styles from './menus.scss';

/**
 * @class MenuDisplay
 */
class MenuDisplay extends Component {
  static propTypes = {
    menus: PropTypes.arrayOf(PropTypes.object).isRequired,
    loggedIn: PropTypes.bool.isRequired
  }

  state = {
    currentMenuIndex: 0
  }

  goToNextMenu = () => {
    const { currentMenuIndex } = this.state;

    if (this.nextExists()) {
      this.flipMenuBody();
      this.setState({ currentMenuIndex: currentMenuIndex + 1 });
    }
  }

  goToPrevMenu = () => {
    const { currentMenuIndex } = this.state;

    if (this.prevExists()) {
      this.flipMenuBody();

      this.setState({ currentMenuIndex: currentMenuIndex - 1 });
    }
  }

  prevExists = () => {
    const { currentMenuIndex } = this.state;

    return currentMenuIndex > 0;
  }

  nextExists = () => {
    const { currentMenuIndex } = this.state;
    const { menus } = this.props;

    return currentMenuIndex < menus.length - 1;
  }

  /**
   * @return {undefined} undefined
   */
  flipMenuBody() {
    this.menuBody.classList.toggle(styles.flipIn);
    setTimeout(() => this.menuBody.classList.toggle(styles.flipIn), 400);
  }

  /**
   * @returns {JSX} React JSX
   */
  render() {
    const { currentMenuIndex } = this.state;
    const { menus, loggedIn } = this.props;

    return isEmpty(menus) ?
      <div className={styles.noMenu}>
        <NoMenu loggedIn={loggedIn} />
      </div> :
      <div className={styles.slideInUp}>
        <MenuNavigator
          prevExists={this.prevExists}
          nextExists={this.nextExists}
          goToNextMenu={this.goToNextMenu}
          goToPrevMenu={this.goToPrevMenu}
        />
        <div ref={el => this.menuBody = el} className={styles.menuBody}>
          <div className={styles.catererName}>
            {menus[currentMenuIndex].caterer.username}&apos;s menu
          </div>
          {
            menus[currentMenuIndex].categories.map(category => (
              <MenuCategory key={category.id} category={category} />
            ))
          }
        </div>
      </div>;
  }
}

export default MenuDisplay;
