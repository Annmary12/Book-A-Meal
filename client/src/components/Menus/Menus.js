import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DocumentTitle from 'react-document-title';
import MenuDisplay from './MenuDisplay.js';
import Loader from '../Loader.js';
import styles from './menus.scss';

/**
 * @class Menus
 */
class Menus extends Component {
  static propTypes = {
    getMenusForTheDay: PropTypes.func.isRequired,
    menus: PropTypes.arrayOf(PropTypes.object).isRequired,
    loading: PropTypes.bool.isRequired,
    loggedIn: PropTypes.bool.isRequired
  }

  /**
   * @returns {undefined} undefined
   */
  componentDidMount() {
    this.props.getMenusForTheDay();
  }

  /**
   * @returns {JSX} React JSX
   */
  render() {
    const { loading, menus, loggedIn } = this.props;

    return (
      <DocumentTitle title="Menus - Book-A-Meal">
        <div className={styles.menu}>
          <h2 className={styles.menuHeader}>~ Menus for the day ~</h2>
          {
            loading ?
              <Loader /> :
              <MenuDisplay
                menus={menus}
                loggedIn={loggedIn}
              />
          }
        </div>
      </DocumentTitle>
    );
  }
}

export default Menus;
