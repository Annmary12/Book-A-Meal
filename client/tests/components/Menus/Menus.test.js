import React from 'react';
import Menus from '../../../src/components/Menus/Menus';

describe('Menus component', () => {
  const props = {
    getMenusForTheDay: jest.fn(),
    menus:[],
    loading: false,
    loggedIn: false
  }

  it('should match snapshot if loading is false', () => {
    const wrapper = shallow(
      <Menus
        {...props}
      />
    );

    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot if loading is true', () => {
    const wrapper = shallow(
      <Menus
        {...props}
        loading={true}
      />
    );

    expect(wrapper).toMatchSnapshot();
  })
});