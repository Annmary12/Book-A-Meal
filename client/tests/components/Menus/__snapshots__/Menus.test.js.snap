// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menus component should match snapshot if loading is false 1`] = `
<div
  className="menu"
>
  <h2
    className="menuHeader"
  >
    ~ Menus for the day ~
  </h2>
  <MenuDisplay
    loggedIn={false}
    menus={Array []}
  />
</div>
`;

exports[`Menus component should match snapshot if loading is true 1`] = `
<div
  className="menu"
>
  <h2
    className="menuHeader"
  >
    ~ Menus for the day ~
  </h2>
  <Loader
    message="loading..."
    width={40}
  />
</div>
`;
